pkg/webhook/managedresource/managedresource_validating_webhook_test.go

```diff
@@ -0,0 +1,222 @@
+			operation:      admissionv1.Update,
+			oldLabels:      map[string]string{"foo": "bar"},
+			oldAnnotations: map[string]string{"baz": "qux"},
+			newLabels:      map[string]string{"foo": "bar"},
+			newAnnotations: map[string]string{"baz": "qux"},
+			expectedResp:   admission.Allowed(""),
+		},
+		{
+			name:         "denied - error on getLabelsAndAnnotations failure",
+			operation:    admissionv1.Create,
+			expectedResp: admission.Errored(http.StatusInternalServerError, fmt.Errorf("object does not implement the Object interfaces")),
+			modReq: func(req *admission.Request) {
+				req.Object = runtime.RawExtension{Object: nil}
+			},
+		},
+		{
+			name:           "denied - managed by arm in labels, not whitelisted",
+			operation:      admissionv1.Create,
+			oldLabels:      nil,
+			oldAnnotations: nil,
+			newLabels:      map[string]string{managedByArmKey: managedByArmValue},
+			newAnnotations: nil,
+			expectedResp:   admission.Denied(fmt.Sprintf(resourceDeniedFormat, metav1.GroupVersionKind{Kind: "TestKind"}, "test-resource", "default")),
+		},
+		{
+			name:           "denied - managed by arm in annotations, not whitelisted",
+			operation:      admissionv1.Update,
+			oldLabels:      nil,
+			oldAnnotations: nil,
+			newLabels:      nil,
+			newAnnotations: map[string]string{managedByArmKey: managedByArmValue},
+			expectedResp:   admission.Denied(fmt.Sprintf(resourceDeniedFormat, metav1.GroupVersionKind{Kind: "TestKind"}, "test-resource", "default")),
+		},
+		{
+			name:           "allowed for other operations",
+			operation:      admissionv1.Connect,
+			oldLabels:      nil,
+			oldAnnotations: nil,
+			newLabels:      nil,
+			newAnnotations: nil,
+			expectedResp:   admission.Allowed(""),
+		},
+		{
+			name:           "allowed for other operations - managed by arm, but user whitelisted",
+			username:       "fleet1p",
+			operation:      admissionv1.Update,
+			oldLabels:      map[string]string{"managedBy": managedByArmValue},
+			oldAnnotations: nil,
+			newLabels:      nil,
+			newAnnotations: nil,
+			expectedResp:   admission.Allowed(""),
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			oldObj := makeUnstructured(tt.oldLabels, tt.oldAnnotations)
+			newObj := makeUnstructured(tt.newLabels, tt.newAnnotations)
+			req := admission.Request{
+				AdmissionRequest: admissionv1.AdmissionRequest{
+					Operation:   tt.operation,
+					Name:        "test-resource",
+					Namespace:   "default",
+					OldObject:   runtime.RawExtension{Object: oldObj},
+					Object:      runtime.RawExtension{Object: newObj},
+					Kind:        metav1.GroupVersionKind{Kind: "TestKind"},
+					RequestKind: &metav1.GroupVersionKind{Kind: "TestKind"},
+				},
+			}
+			req.UserInfo = authenticationv1.UserInfo{
+				Username: tt.username,
+			}
+			if tt.modReq != nil {
+				tt.modReq(&req)
+			}
+			resp := validator.Handle(context.Background(), req)
+			assert.Equal(t, tt.expectedResp.Allowed, resp.Allowed)
```

Comment:
We don't use or avoid using assert packages in the repo please use the cmp package https://github.com/Azure/fleet/blob/main/pkg/controllers/clusterresourceplacement/controller_test.go#L761